File: DSA Unit 06/Static/Q2.cpp
Single File Programming Question
Problem Statement



In a city, there are several locations connected by roads. Each road has a certain distance. You are tasked with finding the shortest distances between all pairs of locations in the city using the Floyd-Warshall algorithm.



Write a program that calculates the shortest path between all pairs of locations in the city.



Example

Input:

4

4

0 1 3

1 2 1

2 3 1

0 2 5



Output:

0 3 4 5 

3 0 1 2 

4 1 0 1 

5 2 1 0



Explanation:

The city has 4 locations (0 to 3).

There are 4 roads connecting the locations.

The roads and their distances are:

0 → 1 with distance 3

1 → 2 with distance 1

2 → 3 with distance 1

0 → 2 with distance 5

Initially, the distance matrix is set to infinity (INF) except for direct roads and diagonal values (0).

The Floyd-Warshall algorithm updates the matrix by checking intermediate paths.

The shortest path from 0 to 2 (via 1) is updated to 4 (0 → 1 → 2) instead of 5.

The shortest path from 0 to 3 is 5 (0 → 1 → 2 → 3).

The final distance matrix is printed, showing the shortest distances.

0 3 4 5

3 0 1 2

4 1 0 1

5 2 1 0

Input format :
The first line contains an integer N, representing the number of locations in the city.

The second line contains an integer M, representing the number of roads in the city.

The next M lines each contain three space-separated integers: u, v, w where:

u and v represent two locations connected by a road.
w represents the distance between u and v.
Output format :
The output prints a matrix, where the value at the i-th row and j-th column represents the shortest distance between location i and location j.



Refer to the sample output for the formatting specifications.

Code constraints :
In this scenario, the given test cases will fall under the following constraints:

2 ≤ N ≤ 5

1 ≤ M ≤ 10

0 ≤ w ≤ 100

Sample test cases :
Input 1 :
4
4
0 1 3
1 2 1
2 3 1
0 2 5
Output 1 :
0 3 4 5 
3 0 1 2 
4 1 0 1 
5 2 1 0 
Input 2 :
3
3
0 1 2
1 2 3
0 2 6
Output 2 :
0 2 5 
2 0 3 
5 3 0 



#include <iostream>
#include <climits>
using namespace std;
#define INF INT_MAX
void floydWarshall(int dist[][5], int N) {
    for (int k = 0; k < N; k++) {
        for (int i = 0; i < N; i++) {
            for (int j = 0; j < N; j++) {
                if (dist[i][k] != INF && dist[k][j] != INF && dist[i][j] > dist[i][k] + dist[k][j])
                    dist[i][j] = dist[i][k] + dist[k][j];
            }
        }
    }
}

void printSolution(int dist[][5], int N) {
    for (int i = 0; i < N; i++) {
        for (int j = 0; j < N; j++) {
            if (dist[i][j] == INF)
                cout << "INF ";
            else
                cout << dist[i][j] << " ";
        }
        cout << endl;
    }
}

int main() {
    int N, M;
    cin >> N >> M;
    int dist[5][5];
    for (int i = 0; i < N; i++) {
        for (int j = 0; j < N; j++) {
            if (i == j)
                dist[i][j] = 0;
            else
                dist[i][j] = INF;
        }
    }
    for (int i = 0; i < M; i++) {
        int u, v, w;
        cin >> u >> v >> w;
        dist[u][v] = dist[v][u] = (dist[u][v] < w) ? dist[u][v] : w;
    }
    floydWarshall(dist, N);
    printSolution(dist, N);
    return 0;
}

File: Java/Unit 02/Lec 12/Q2.java
Single File Programming Question
Problem Statement



Hari is working on a Java program for his homework to calculate the sum of two and three integers.
  Write a program that allows Hari to input two integers and three integers separately.
  Use overloaded methods called sum to calculate and display the sum for each case.

Input format :
The first line of input consists of two space-separated integers m and n.

The second line consists of three space-separated integers a, b, and c.

Output format :
The first line of output prints "Sum of two integers: " followed by an integer representing the sum of m and n.

The second line prints "Sum of three integers: " followed by an integer representing the sum of a, b, and c.



Refer to the sample output for formatting specifications.

Code constraints :
The given test cases fall under the following constraints:

1 ≤ input integers ≤ 1000

Sample test cases :
Input 1 :
12 34 
56 78 63
Output 1 :
Sum of two integers: 46
Sum of three integers: 197
Input 2 :
523 178 
481 989 124
Output 2 :
Sum of two integers: 701
Sum of three integers: 1594




  import java.util.Scanner;

class Summation {

    int sum(int m, int n) {
        return m + n;
    }

    int sum(int a, int b, int c) {
        return a + b + c;
    }
}

public class Main {

    public static void main(String[] args) {

        Scanner scanner = new Scanner(System.in);

        int m = scanner.nextInt();
        int n = scanner.nextInt();
        int a = scanner.nextInt();
        int b = scanner.nextInt();
        int c = scanner.nextInt();

        Summation summation = new Summation();

        int sum2 = summation.sum(m, n);
        int sum3 = summation.sum(a, b, c);

        System.out.println("Sum of two integers: " + sum2);
        System.out.println("Sum of three integers: " + sum3);

        scanner.close();
    }
}
